src/AWSCredentials.jl

```diff
@@ -24,75 +27,114 @@ The fields `access_key_id` and `secret_key` hold the access keys used to authent
 
 [Temporary Security Credentials](http://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_temp.html) require the extra session `token` field.
 
-
 The `user_arn` and `account_number` fields are used to cache the result of the [`aws_user_arn`](@ref) and [`aws_account_number`](@ref) functions.
-
-The `AWSCredentials()` constructor tries to load local Credentials from
-environment variables, `~/.aws/credentials`, `~/.aws/config` or EC2 instance credentials.
-To specify the profile to use from `~/.aws/credentials`, do, for example, `AWSCredentials(profile="profile-name")`.
 """
 mutable struct AWSCredentials
     access_key_id::String
     secret_key::String
     token::String
     user_arn::String
     account_number::String
+    expiry::DateTime
 
-    function AWSCredentials(access_key_id, secret_key,
-                            token="", user_arn="", account_number="")
-        new(access_key_id, secret_key, token, user_arn, account_number)
+    function AWSCredentials(access_key_id,secret_key,
+                            token="", user_arn="", account_number="";
+                            expiry=now(UTC) + DEFAULT_DURATION)
```

Comment:
I know that boto uses an estimated expiry time to do this but I'm not certain it is the best approach. What if we instead renewed when the credential is expired. AWS gives a pretty clear error message when this occurs:
```julia
julia> AWSSDK.STS.get_caller_identity()
ERROR: ExpiredToken -- The security token included in the request is expired
HTTP.ExceptionRequest.StatusError(403, HTTP.Messages.Response:
"""
HTTP/1.1 403 Forbidden
x-amzn-RequestId: d940c825-263e-11e9-9a6f-47eada21823e
Content-Type: application/json
Content-Length: 166
Date: Fri, 01 Feb 2019 16:31:38 GMT

{"Error":{"Code":"ExpiredToken","Message":"The security token included in the request is expired","Type":"Sender"},"RequestId":"d940c825-263e-11e9-9a6f-47eada21823e"}""")
```

Reply:
I think it is the right thing to do. It would be nice to avoid error messages for normal behaviour.

Reply:
I was thinking we would retry automatically. I think approach is beyond the scope of the changes here though.